### PR TITLE
Barchart: Ensure data field overrides apply

### PR DESCRIPTION
**Pull request by drew08t**
What this PR does / why we need it:
Currently when hovering over columns in Bar Chart, all data links will display regardless of field overrides being applied.

- [x] Change displayValues structure to an array of objects
- [x] Ensure highlight styling is working correctly
- [x] Verify that other panels using this hover view are behaving as expected, especially Geomap

Before:
May-04-2023 13-03-24

After:
May-04-2023 12-58-42
Highlighting: 
May-08-2023 17-30-29

Closes https://github.com/grafana/support-escalations/issues/5613

### CODE REVIEW

File: public/app/plugins/panel/geomap/components/DataHoverView.tsx [hunk 1]
```diff
@@ -24,27 +24,41 @@ export interface Props {
   header?: string;
 }
 
+//string, unknown, string, boolean
```
Comment (nmarrs) [inline public/app/plugins/panel/geomap/components/DataHoverView.tsx:27]:
```suggestion
```

File: public/app/plugins/panel/geomap/components/DataHoverView.tsx [hunk 2]
```diff
@@ -24,27 +24,41 @@ export interface Props {
...
+
 export const DataHoverView = ({ data, rowIndex, columnIndex, sortOrder, mode, header = undefined }: Props) => {
   const styles = useStyles2(getStyles);
 
   if (!data || rowIndex == null) {
     return null;
   }
-
+  const fields = data.fields.map((f, idx) => {
+    return { ...f, highlight: idx === columnIndex };
```
Comment (nmarrs) [inline public/app/plugins/panel/geomap/components/DataHoverView.tsx:42]: maybe we should name added property to field something a bit more descriptive?

maybe something like `isFieldHovered` - along those lines of the what vs. an effect of the property down the line?

File: public/app/plugins/panel/geomap/components/DataHoverView.tsx [hunk 3]
```diff
@@ -24,27 +24,41 @@ export interface Props {
...
     return null;
   }
 
-  const displayValues: Array<[string, unknown, string]> = [];
+  const displayValues: DisplayValue[] = [];
   const links: Array<LinkModel<Field>> = [];
   const linkLookup = new Set<string>();
 
-  for (const f of orderedVisibleFields) {
+  for (let i = 0; i < orderedVisibleFields.length; i++) {
```
Comment (nmarrs) [inline public/app/plugins/panel/geomap/components/DataHoverView.tsx:57]: probably can switch this back to the other loop type 🤷‍♂️

File: public/app/plugins/panel/geomap/components/DataHoverView.tsx [hunk 4]
```diff
@@ -24,27 +24,41 @@ export interface Props {
   header?: string;
 }
 
+//string, unknown, string, boolean
+interface DisplayValue {
+  name: string;
+  value: unknown;
```
Comment (nmarrs) [inline public/app/plugins/panel/geomap/components/DataHoverView.tsx:30]: should this type be unknown? do we have a type in grafana data for the union of possible value types from a field?

File: public/app/plugins/panel/geomap/components/DataHoverView.tsx (hunk 1 above)
Comment (drew08t) [reply public/app/plugins/panel/geomap/components/DataHoverView.tsx:27]: Whoops, good catch.

**Review by nmarrs — APPROVED** [review]
🍰

**Review by nmarrs — APPROVED** [review]
🍰

File: public/app/plugins/panel/geomap/components/DataHoverView.tsx [hunk 5]
```diff
@@ -96,19 +113,14 @@ export const DataHoverView = ({ data, rowIndex, columnIndex, sortOrder, mode, he
...
-                <td>{renderValue(v[2])}</td>
-              </tr>
-            ))}
-          {mode === TooltipDisplayMode.Single && columnIndex && (
-            <tr key={`${columnIndex}/${rowIndex}`}>
-              <th>{displayValues[columnIndex][0]}:</th>
-              <td>{renderValue(displayValues[columnIndex][2])}</td>
+          {displayValues.map((v, i) => (
+            <tr key={`${i}/${rowIndex}`}>
+              <div className={v.highlight ? styles.highlight : ''}>
```
Comment (drew08t) [inline public/app/plugins/panel/geomap/components/DataHoverView.tsx:118]: It looks like this little styling workaround is breaking the build.

 Warning: validateDOMNesting(...): <td> cannot appear as a child of <div>.
          at td
          at div
          at tr
          at tbody
          at table
          at div

**Comment by nmarrs** [comment]
This should be mergeable once https://github.com/grafana/grafana/pull/68276 is approved and this is merged onto latest main 👍

**Comment by marinnedea** [comment]
Hi team, in which Grafana versions is this pushed? I just tested with 9.2.1 and 9.5.2 and  I still see all Data Links on all bars..

**Comment by adela-almasan** [comment]
Hi @marinnedea, this fix should be in v10.0.x - it was not back-ported to earlier versions.

**Comment by gtk-grafana** [comment]
This PR introduced a regression @drew08t 
https://github.com/grafana/grafana/issues/69578

**Pull request merged**